src/plugins/intel_npu/tests/functional/behavior/infer_request_run.hpp

```diff
@@ -1828,6 +1828,118 @@ TEST_P(CpuVaTensorsTests, checkResultsAfterStateTensorsUseImportCpuVa1) {
+}
+
+TEST_P(CpuVaTensorsTests, SetAlignedDifferentTensorTwiceForSameInferRequestsNoThrow) {
+    SKIP_IF_CURRENT_TEST_IS_DISABLED();
+
+    const auto shape = ov::PartialShape{1, 16, 16, 16};
+    auto model = createModel(element::f32, shape, "N...");
+    auto compiledModel = core->compile_model(model, target_device, configuration);
+
+    const auto elementType = compiledModel.inputs()[0].get_element_type();
+    const size_t byteSize = ov::shape_size(shape.get_shape()) * elementType.size();
+
+    auto inferRequest = compiledModel.create_infer_request();
+    float* data = static_cast<float*>(::operator new(byteSize, std::align_val_t(4096)));
+    auto tensor1 = ov::Tensor{elementType, shape.get_shape(), data};
+    auto tensor2 = ov::Tensor{elementType, shape.get_shape(), data};
+    OV_ASSERT_NO_THROW(inferRequest.set_tensor(compiledModel.inputs()[0], tensor1));
+    OV_ASSERT_NO_THROW(inferRequest.set_tensor(compiledModel.inputs()[0], tensor2));
+
+    OV_ASSERT_NO_THROW(inferRequest.infer());
+    ::operator delete(data, std::align_val_t(4096));
+}
+
+TEST_P(CpuVaTensorsTests, SetAlignedSameTensorForSameInferRequestsNoThrow) {
+    SKIP_IF_CURRENT_TEST_IS_DISABLED();
+
+    const auto shape = ov::PartialShape{1, 16, 16, 16};
+    auto model = createModel(element::f32, shape, "N...");
+    auto compiledModel = core->compile_model(model, target_device, configuration);
+
+    const auto elementType = compiledModel.inputs()[0].get_element_type();
+    const size_t byteSize = ov::shape_size(shape.get_shape()) * elementType.size();
+
+    auto inferRequest = compiledModel.create_infer_request();
+    float* data = static_cast<float*>(::operator new(byteSize, std::align_val_t(4096)));
+    auto tensor = ov::Tensor{elementType, shape.get_shape(), data};
+    OV_ASSERT_NO_THROW(inferRequest.set_tensor(compiledModel.inputs()[0], tensor));
+    OV_ASSERT_NO_THROW(inferRequest.set_tensor(compiledModel.inputs()[0], tensor));
+
+    OV_ASSERT_NO_THROW(inferRequest.infer());
+    ::operator delete(data, std::align_val_t(4096));
+}
+
+TEST_P(CpuVaTensorsTests, SameAlignedPtrForTwoInferRequestNoThrow) {
+    SKIP_IF_CURRENT_TEST_IS_DISABLED();
+
+    const auto shape = ov::PartialShape{1, 16, 16, 16};
+    auto model = createModel(element::f32, shape, "N...");
+    auto compiledModel = core->compile_model(model, target_device, configuration);
+
+    const auto elementType = compiledModel.inputs()[0].get_element_type();
+    const size_t byteSize = ov::shape_size(shape.get_shape()) * elementType.size();
+
+    auto inferRequest1 = compiledModel.create_infer_request();
+    auto inferRequest2 = compiledModel.create_infer_request();
+    float* data = static_cast<float*>(::operator new(byteSize, std::align_val_t(4096)));
+    auto tensor = ov::Tensor{elementType, shape.get_shape(), data};
+    OV_ASSERT_NO_THROW(inferRequest1.set_tensor(compiledModel.inputs()[0], tensor));
+    OV_ASSERT_NO_THROW(inferRequest2.set_tensor(compiledModel.inputs()[0], tensor));
```

Comment:
So we want to support two import tensor for same data ptr, or the first inferrequest to use import tensor, but the second one to use local L0 tensor?